### PR TITLE
feat [IDP-1867] : Added the docs on usage of layout page

### PR DESCRIPTION
**Pull request by Debanitrkl**
# Harness Developer Pull Request
Added the docs on usage of layout page

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: Usage of Layout in IDP Admin
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:2]: `Configuring Layout of IDP Catalog pages`

might be more meaningful maybe

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+sidebar_position: 100
+helpdocs_topic_id:
+helpdocs_category_id:
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
+## Introduction
+
+In Harness IDP we take care of the UI configuration for your plugins and core-features, unlike that of Backstage where you need to update the entity pages in the app. But whle doing so we don't take away your ability to configure the cards and tabs instead we auto-ingest most of the values once the plugins are enabled and those values can be further configured according to your needs.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:13]:
```suggestion
In Harness IDP we take care of the UI configuration for your plugins and core-features, unlike that of Backstage where you need to update the entity pages in the app. But while doing so we don't take away your ability to configure the catalog cards, tabs and side bar instead we auto-ingest most of the values once the plugins are enabled and those values can be further configured according to your needs. 
```

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+helpdocs_is_published: true
+---
+
+## Introduction
+
+In Harness IDP we take care of the UI configuration for your plugins and core-features, unlike that of Backstage where you need to update the entity pages in the app. But whle doing so we don't take away your ability to configure the cards and tabs instead we auto-ingest most of the values once the plugins are enabled and those values can be further configured according to your needs. 
+
+## Layout Page
+
+The **Layout** page is under the **Admin** section and is a comprehensive and modular interface designed for plugins as well as core features, leveraging HTML for a dynamic and interactive user experience. This document provides detailed information about the layout, its components, and usage.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:17]:
```suggestion
The **Layout** page is under the **Admin** section and is a comprehensive and modular interface designed for plugins as well as core features, leveraging YAML for a dynamic and interactive user experience. This document provides detailed information about the layout, its components, and usage.
```

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+- Title: Overview
+- Components:
+    - `EntityAboutCard`: Provides a brief about the entity.
+    - `EntityOrphanWarning`: Displays warnings for orphan entities.
+    - `EntityProcessingErrorsPanel`: Shows processing errors in a panel layout.
+- Props: variant: gridItem
+- GridProps: md: 6
+- `EntityScoreCard`: Displays scores and metrics related to the entity.
+- GridProps: md: 6
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:37]:
```suggestion
- Title: Overview
- Components:
    - `EntityAboutCard`: Provides a brief about the entity.
    - `EntityOrphanWarning`: Displays warnings for orphan entities.
    - `EntityProcessingErrorsPanel`: Shows processing errors in a panel layout.
- Props: variant: gridItem
- GridProps: md: 6
- `EntityScoreCard`: Displays scores and metrics related to the entity.
- GridProps: md: 6
```

this should be in code blocks I believe using ```

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+### CI/CD Tab
+- Path: /ci-cd
+- Title: CI/CD
+- Components:
+- `EntitySwitch`: Dynamically displays CI/CD information based on service availability.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:43]: code blocks and indentation

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+```html
+<EntityLayout>
+  <!-- Overview Tab -->
+  <Tab path="/" title="Overview">
+    <EntityAboutCard />
+    <!-- Additional components -->
+  </Tab>
+  <!-- Additional tabs -->
+</EntityLayout>
+```
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:66]: is HTML supported as well? I didn't know @jenil-harness

Comment (jenil-harness) [reply docs/internal-developer-portal/get-started/layout-deepdive.md:66]: No. HTML/JSX is not supported.

File: docs/internal-developer-portal/get-started/layout-deepdive.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+### CI/CD Tab
+- Path: /ci-cd
+- Title: CI/CD
+- Components:
+- `EntitySwitch`: Dynamically displays CI/CD information based on service availability.
+
+### Additional Tabs
+API, Deps, Scorecard, TechDocs, EntityGithubPullRequestsContent, EntityKubernetesContent, DynatraceTab: Each of these tabs follows a similar structure, containing components specific to their functionality.
+
+## Understanding md and xs Units in Grid Layouts
```
Comment (jenil-harness) [inline docs/internal-developer-portal/get-started/layout-deepdive.md:48]: Let's not add xs units as Harness is not supported on xs devices.

Better we add md.
The entire page is divided into 12 grids.
md = 1 means 1/12 of the available space

**Review by jenil-harness — APPROVED** [review]
LGTM

**Pull request merged**